### PR TITLE
Add CO Page for Dashboard

### PR DESCRIPTION
**Pull request by iszmais**
https://docu.ilias.de/goto_docu_wiki_wpage_8083_1357.html

### CODE REVIEW

**Comment by iszmais** [comment]
@lscharmer pls have a look into this. Unfortunatly im forced to follow some workflows for that feature but im glad about every improvment recommendation thats within range.

File: Services/Dashboard/classes/class.ilDashboardGUI.php [hunk 1]
```diff
@@ -248,11 +263,22 @@ public function show(): void
...
-        $this->tpl->setTitle($this->lng->txt("dash_dashboard"));
-        $this->tpl->setTitleIcon(ilUtil::getImagePath("standard/icon_dshs.svg"), $this->lng->txt("dash_dashboard"));
         $this->tpl->setVariable("IMG_SPACE", ilUtil::getImagePath("media/spacer.png"));
 
-        $this->tpl->setContent($this->getCenterColumnHTML());
+        $content = $this->getCenterColumnHTML();
+        if(ilPageObject::_exists(ilDashboardPage::PARENT_TYPE, ilDashboardPage::ID, $this->user->getLanguage())) {
+            $content = (new ilDashboardPageGUI(ilDashboardPage::PARENT_TYPE, 0))->showPage() . $content;
+        } elseif(ilPageObject::_exists(ilDashboardPage::PARENT_TYPE, ilDashboardPage::ID, $this->lng->getDefaultLanguage())) {
+            $content = (new ilDashboardPageGUI(ilDashboardPage::PARENT_TYPE, 0))->showPage() . $content;
```
Comment (lscharmer) [inline Services/Dashboard/classes/class.ilDashboardGUI.php]: Please remove the duplicated code.

**Review by lscharmer — APPROVED** [review]

**Comment by iszmais** [comment]
@lscharmer pls have another look at this since i change some major behavior.

File: Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        global $DIC;
+        $this->dic = $DIC;
+        $this->page = new ilDashboardPage();
+    }
+
+    public function executeCommand(): void
+    {
+        if ($this->dic->ctrl()->getNextClass() === strtolower(ilDashboardPageGUI::class)) {
+            $lang = $this->dic->http()->request()->getQueryParams()[$this->page->getParentType() . '_lang'];
```
Comment (lscharmer) [inline Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php]: Why are you not using the `$this->dic->http()->wrapper()->query()`?

File: Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            if (is_array($lang)) {
+                $lang = reset($lang);
```
Comment (lscharmer) [inline Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php]: Why do you need to reset the array. If you want the first element use `current` instead, as `reset` is used for a different purpose.

**Review by lscharmer — CHANGES_REQUESTED** [review]
Hi @iszmais,
thank you for this PR. There are just 2 minor points that need to be changed / answered.

Best regards
@lscharmer

File: Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php (hunk 3 above)
Comment (iszmais) [reply Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php]: I switch to a proper request wrapper therefore this is no longer needed in any way.

**Review by lscharmer — APPROVED** [review]

**Pull request closed without merging**